jsFiles/app.js

```diff
@@ -34,7 +37,12 @@ form.addEventListener("submit", search);
 // submit event works by pressing the button or pressing enter after making an input!
 
 //------------FETCHING WEATHER DATA FROM OPEN WEATHER API---------------------------
-function fetchData(apiUrl) {
+/**
+ * @param {string} apiUrl - url
+ * @param {boolean} searchCity - default value is false, true when the fetchData
+ * is called from the search function.
+ */
+function fetchData(apiUrl, searchCity = false) {
```

Comment:
The way it is working up to here is that the page loads and displays the date and time of the user's location. That happens only if the user allows the geolocation in the browser. Then the user can search for the weather of another city. 
When the users search the function search() gets trigger and only then the time of the city searched gets displayed, but it doesn't overwrite the user's time and date location display element, it adds a new element on the webpage. So there is some code that I like to run only if the a city is search.... in the search function I call fetchData(api, true); with an extra parameter.
and on fetchData() I need it to run this code. Which gives me the time.
<code>
 if (searchCity == true) {
        searchLocationTimeElement.innerHTML = cityTime(
          currentDateUserLocation,
          offsetTime
        );
</code>
The reason this code runs from fetchData is because I need the offsetTime that I collect from the API. The offsetTime is a new (it is introduced in this pull request) collected value for my program.